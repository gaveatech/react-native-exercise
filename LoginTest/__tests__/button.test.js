import React from 'react';
import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react-native';

import Btn from '../src/components/Button';

describe('<Button>', () => {
  it('Snapshot', () => {
    const tree = renderer.create(<Btn title="Botao" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Button press', () => {
    const pressBtn = jest.fn();
    const {getByTestId} = render(<Btn title="Botao" onPress={pressBtn} />);

    fireEvent.press(getByTestId('touch'));
    expect(getByTestId('touch')).toBeTruthy();
    expect(pressBtn).toHaveBeenCalledTimes(1);
  });
});
