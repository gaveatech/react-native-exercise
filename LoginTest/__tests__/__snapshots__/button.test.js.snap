// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> Snapshot 1`] = `
<View
  accessible={true}
  bgColor="#000"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="100%"
  style={
    Array [
      Object {
        "backgroundColor": "#000",
        "borderRadius": 20,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 15,
        "paddingBottom": 10,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 10,
        "width": "100%",
      },
    ]
  }
  testID="touch"
>
  <Text
    color="#fff"
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "RobotoRegular",
          "fontSize": 16,
          "textAlign": "center",
        },
      ]
    }
  >
    Botao
  </Text>
</View>
`;
